File: libraries/uava/src/main/java/com/example/uava/Preconditions.java
package com.example.uava;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

public final class Preconditions {
	private Preconditions() {
	}

	@NonNull
	public static <T> T checkNotNull(T reference) {
		if (reference == null) {
			throw new NullPointerException();
		}

		return reference;
	}

	@NonNull
	public static <T> T checkNotNull(T reference, @Nullable Object errorMessage) {
		if (reference == null) {
			throw new NullPointerException(String.valueOf(errorMessage));
		}

		return reference;
	}

	@NonNull
	public static String checkNotNullOrEmpty(String value, @Nullable String errorMessage) {
		if (value == null || value.isEmpty()) {
			throw new IllegalArgumentException(errorMessage);
		}

		return value;
	}

	public static void checkArgument(boolean expression, @Nullable Object errorMessage) {
		if (!expression) {
			throw new IllegalArgumentException(String.valueOf(errorMessage));
		}
	}
}

File: libs/toolkit/src/main/kotlin/br/com/vitorsalgado/example/toolkit/Preconditions.kt
package br.com.vitorsalgado.example.toolkit

object Preconditions {
  fun <T> checkNotNull(reference: T?): T {
    if (reference == null) {
      throw NullPointerException()
    }

    return reference
  }

  fun <T> checkNotNull(reference: T?, errorMessage: Any?): T {
    if (reference == null) {
      throw NullPointerException(errorMessage.toString())
    }

    return reference
  }

  fun checkNotNullOrEmpty(value: String?, errorMessage: String?): String {
    if (value == null || value.isEmpty()) {
      throw IllegalArgumentException(errorMessage)
    }

    return value
  }

  fun checkArgument(expression: Boolean, errorMessage: Any?) {
    if (!expression) {
      throw IllegalArgumentException(errorMessage.toString())
    }
  }
}
